public/js/common.js

```diff
@@ -3,6 +3,41 @@
  */
 const LRR = {};
 
+// This class is used to wrap URLs that point into the app, including
+// API endpoints and browser targets. It reads the configured base URL
+// from the template, then prepends it to the provided URL in the
+// toString() method.
+
+// Every operation involving an app-internal URL should be wrapped in
```

Comment:
Is there a way we could potentially enforce this thru linter rules?  
I'm aware it's potentially impossible since we can't always know which URLs are internal and which aren't, but I thought I'd ask regardless.  

This class looks great to me but I am worried that it's easy to forget about 🤔

Reply:
I would love it if this were possible; you're right that this is easy to forget about. Unfortunately, I can't think of an easy way to do it. The app generates URLs in all sorts of ways from all sorts of places, including in inline HTML inside string constants. Most of those are path-only URLs that aren't necessarily programmatically distinguishable from any other string.

I don't know how a linter would even be able to determine what strings are URLs and what aren't, leaving aside sorting them into internal vs. external URLs.

Reply:
Yup, fair enough!